@grafana/ui: RadioButtonGroup documentation

jessover9000 opened the pull request:
This is a docs update. Added info about how RadioButtonGroups generally work and when to use another component instead.

jessover9000 committed:
Add "When to use" info to RadioButtonGroup.mdx

jessover9000 committed:
Clarify more clearly when to use 'Select' instead

sarlinska commented on packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx:
```diff
@@ -5,9 +5,13 @@ import { RadioButtonGroup } from './RadioButtonGroup';
-Use `RadioButtonGroup` if there are up to four options available. Otherwise use Select component.
+### When to use
+
+Use `RadioButtonGroup` for mutually exclusive selections if there are up to four options available. For a mutually exclusive selection of more than four options, use `Select` component. 
```
Why have we decided it's "up to four options"? I am not saying it should be less or more just trying to understand the reasoning.

tskarhed replied:
I think the reasoning is a matter of both width of the RadioButtonGroup and avoiding clutter.

sarlinska commented on packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.mdx:
```diff
@@ -5,9 +5,13 @@ import { RadioButtonGroup } from './RadioButtonGroup';
+
+Use `RadioButtonGroup` for mutually exclusive selections if there are up to four options available. For a mutually exclusive selection of more than four options, use `Select` component. 
+
+Like `Checkbox`, `RadioButtonGroup` only triggers changes on sending a form. Furthermore, radio buttons can only exist in this group. If you want one single option, or if you want your component to trigger changes immediately, it's better to use `Switch` instead. To offer multiple choices within the same group or context which are not mutually exclusive, use `Checkbox` instead.
```
I am not sure "triggering changes on sending a form" will be true for example in Panel Edit. @torkelo ?

tskarhed replied:
I agree that this isn't something form exclusive

jessover9000 committed:
Update RadioButtonGroup.mdx

Add line explaining why we want select for more than four options instead of a radio button group

jessover9000 committed:
Update RadioButtonGroup.mdx

remove part about when elements are triggering changes.

tskarhed committed:
Merge branch 'master' into jessover9000/docs-radiobuttongroup
